core/server/models/post.js

```diff
@@ -119,13 +119,24 @@ Post = ghostBookshelf.Model.extend({
             }
         });
 
-        this.on('destroying', function (model/*, attr, options*/) {
-            return model.load('tags').call('related', 'tags').call('detach').then(function then() {
-                if (model.previous('status') === 'published') {
-                    model.emitChange('unpublished');
-                }
-                model.emitChange('deleted');
-            });
+        this.on('destroying', function (model, options) {
+            return model.load('tags', options)
+                .then(function (response) {
+                    if (!response.related || !response.related('tags') || !response.related('tags').length) {
+                        return;
+                    }
+
+                    return Promise.mapSeries(response.related('tags').models, function (tag) {
+                        return baseUtils.tagUpdate.detachTagFromPost(model, tag, options)();
+                    });
+                })
+                .then(function () {
+                    if (model.previous('status') === 'published') {
+                        model.emitChange('unpublished');
+                    }
+
+                    model.emitChange('deleted');
+                });
```

Comment:
This is a big change that doesn't seem to be explained in the PR message. Adding explanations in the commit message is nice, but in the PR message it is needed so that I am able to understand what you are changing and why please :)